### PR TITLE
Update verification

### PR DESCRIPTION
**Pull request by killerswin2**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    Uses the new 2.10 compatibleItems command and compatibleMagazines. The commands are faster in certain mods.

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

File: A3A/addons/core/functions/Templates/Verification/fn_TV_verifyLoadout.sqf [hunk 1]
```diff
@@ -76,87 +96,70 @@ private _getCompatibleAttachements = {
...
-        _compatibleMuzzles append ([_weapon, "muzzle"] call CBA_fnc_compatibleItems);
-    };
+    private _compatibleMuzzles = (compatibleItems  _weapon) select { MUZZLE_TYPE == getNumber ( configFile >> "CfgWeapons" >> _x >> "itemInfo" >> "type" ) };
 
     if !(_muzzle in _compatibleMuzzles) exitWith {
         _invalidReasons pushBack ("Muzzle: "+_muzzle+" is incompatible with "+_weapon+" | Comaptible muzzles: "+ str _compatibleMuzzles);
+
+        if (MUZZLE_TYPE != getNumber ( configFile >> "CfgWeapons" >> _muzzle >> "itemInfo" >> "type" )) then {
+            _invalidReasons pushBack ("Optic: trying to add "+_muzzle+" | It is not an muzzle, verify if its in right slot");
+        };
```
Comment (jaj22) [inline A3A/addons/core/functions/Templates/Verification/fn_TV_verifyLoadout.sqf:111]: I don't think you need the select filter, and it's by far the slowest operation here. Can't you just do two checks:
1. Whether `_muzzle` has the correct item type.
2. Whether it's in `compatibleItems _weapons`.

Comment (killerswin2) [reply A3A/addons/core/functions/Templates/Verification/fn_TV_verifyLoadout.sqf:111]: that was done for cup

Comment (killerswin2) [reply A3A/addons/core/functions/Templates/Verification/fn_TV_verifyLoadout.sqf:111]: huh, I see it now

File: A3A/addons/core/functions/Templates/Verification/fn_TV_verifyLoadout.sqf [hunk 2]
```diff
@@ -1,33 +1,53 @@
...
+        _binocular - Binocular "weapon" array
+        _linkedItems - Linked items
+
+    Return Value:
+        Loadout valid
+        Invalid reasons
+
+    Scope: 
+        Any
+
```
Comment (jaj22) [inline A3A/addons/core/functions/Templates/Verification/fn_TV_verifyLoadout.sqf:28]: Are we really doing this three lines for one word thing? I don't feel like I agreed to that.

Comment (killerswin2) [reply A3A/addons/core/functions/Templates/Verification/fn_TV_verifyLoadout.sqf:28]: not really, it's a hold over from my other pr

File: A3A/addons/core/functions/Templates/Verification/fn_TV_verifyLoadout.sqf [hunk 3]
```diff
@@ -166,52 +172,55 @@ private _validateWeaponMagazine = { //valid class and compatible with weapon and
...
     if (_baseWeapon isEqualTo "") exitWith {true};
-
-    ["CfgWeapons",_baseWeapon] call _validClassCaseSensitive
-    && { [_baseWeapon, _muzzle] call _validMuzzle }
-    && { [_baseWeapon, _rail] call _validRail }
-    && { [_baseWeapon, _optic] call _validOptic }
-    && { [_baseWeapon, _priMag] call _validateWeaponMagazine }
-    && { [_baseWeapon, _secMag] call _validateWeaponMagazine }
-    && { [_baseWeapon, _bipod] call _validBipod };
+    _compatibleItemsWeapon = (compatibleItems _baseWeapon);
```
Comment (HakonRydland) [inline A3A/addons/core/functions/Templates/Verification/fn_TV_verifyLoadout.sqf:198]: add parent variable dependency comment to the functions bellow

**Review by HakonRydland — APPROVED** [review]
minor note, but can still be merged without

**Comment by Bob-Murphy** [comment]
@killerswin2 merge conflict

**Comment by killerswin2** [comment]
@Bob-Murphy Resolved

**Pull request merged**